Unretirerequests

tyll opened the pull request:
This adds initial unretire support. It is currently mainly useful to do the checks needed for unretirement and it shows hints about what to do next. Eventually it should actually do the unretirement itself, but this needs some for checks for figure out what needs to be done, e.g. which koji tag to unblock it.

pypingou commented on pkgdb2client/admin.py:
```diff
@@ -413,6 +406,49 @@ def __handle_request_branch(actionid, action):
+    action['info']['pkg_poc'] = action['user']
+    #  FIXME utils.check_package_creation() also tries to get
+    #  pkg_namespace, but this does not seemt to be present in the
+    #  action object
```
It should be present on recent requests, is it not? Or maybe it's missing to request to unretire (vs request to create), we should fix it if that's the case.

tyll replied:
It only contains the review url in the info object: https://github.com/fedora-infra/pkgdb2/blob/master/pkgdb2/lib/__init__.py#L2014

pypingou replied:
Hm indeed, but then the namespace should be accessible via ``action['package'].get('namespace', 'rpms')``, which I know see won't work either. Let me fix pkgdb2 for this.

pypingou replied:
Should we include: ``action['package'].get('namespace', 'rpms')`` already so that when pkgdb2 is fixed, this will be as well?

pypingou replied:
FTR pkgdb2 is being fixed at: action['package'].get('namespace', 'rpms')

pypingou replied:
Thoughts on this?

tyll replied:
I merged the pull request and updated this PR to use the namespace from the action.

pypingou commented on pkgdb2client/admin.py:
Should we keep the ``()`` around the ``confirm with any key`` as we have above?

pypingou commented on pkgdb2client/admin.py:
```diff
@@ -461,6 +497,8 @@ def do_process(args):
+        elif action['action'] == 'request.unretire':
+            data = __handle_request_unretire(actionid, action)
         else:
             print('Action %s not supported by pkgdb-cli' % action['action'])
```
I'm realizing that the message is wrong here, since it's ``pkgdb-admin`` not ``pkgdb-cli`` :)

tyll replied:
I will address this

tyll commented on pkgdb2client/admin.py:
yes, good idea.

tyll commented:
I addressed the issues you found and some other things. The details in the info object could be changed in pkdb2 and then removed here, but it should not hurt right now.

pypingou commented on pkgdb2client/admin.py:
```diff
@@ -318,34 +344,20 @@ def __handle_request_package(actionid, action):
-                'message': message,
-            }
-        )
+        data, message = deny_action(actionid)
```
Do we do anything with these?

tyll replied:
The message is sent to the bug report, the variable data is returned by handle function.

pypingou replied:
We could use something like ``_, message = deny_action(actionid)`` then to avoid an un-used variable

tyll replied:
good idea, I adjusted it.

tyll replied:
at this point both variables are actually used, but I changed it in line 420 (see diff below)

pypingou commented on pkgdb2client/admin.py:
```diff
@@ -404,27 +416,10 @@ def __handle_request_branch(actionid, action, package):
-            }
-        )
-
+        data, message = deny_action(actionid)
```
same question here

tyll replied:
here only data is used as return value for the handle function.

pypingou commented:
Looks all good to me

tyll commented:
Thank you
